src/error.rs

```diff
@@ -22,3 +23,37 @@ impl From<io::Error> for Error {
         Error::IOError(e)
     }
 }
+
+impl Display for Error {
+    fn fmt(&self, fmt: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            Error::IOError(e) => e.fmt(fmt),
+            Error::LZMAError(e) | Error::XZError(e) => e.fmt(fmt),
```

Comment:
The original error type (io, lzma or xz) should be formatted as well.

Reply:
> 
> 
> I think it would also be useful to implement the [`std::error::Error` trait](https://doc.rust-lang.org/std/error/trait.Error.html) (essentially implementing the `source()` method to return `Some(e)` in case of IO error and `None` otherwise).

I was thinking the same. See the latest commit.